tokio/src/sync/mpsc/chan.rs

```diff
@@ -234,7 +239,11 @@ where
     S: Semaphore,
 {
     fn drop(&mut self) {
-        self.inner.semaphore.drop_permit(&mut self.permit);
+        let notify = self.inner.semaphore.drop_permit(&mut self.permit);
+
+        if notify && self.inner.semaphore.is_idle() {
```

Comment:
Does `is_idle` mean that no permits are currently taken?

Reply:
Correct